pkg/object/objmetadata.go

```diff
@@ -91,15 +110,48 @@ func validateNameChars(name string) bool {
 // Returns an error if unable to parse and create the ObjMetadata
 // struct.
 func ParseObjMetadata(inv string) (ObjMetadata, error) {
-	parts := strings.Split(inv, fieldSeparator)
-	if len(parts) == 4 {
-		gk := schema.GroupKind{
-			Group: strings.TrimSpace(parts[2]),
-			Kind:  strings.TrimSpace(parts[3]),
-		}
-		return CreateObjMetadata(parts[0], parts[1], gk)
+	fields, err := parseObjFields(inv)
+	if err != nil {
+		return ObjMetadata{}, err
+	}
+	gk := schema.GroupKind{
+		Group: strings.TrimSpace(fields[2]),
+		Kind:  strings.TrimSpace(fields[3]),
 	}
-	return ObjMetadata{}, fmt.Errorf("unable to decode inventory: %s", inv)
+	return CreateObjMetadata(fields[0], fields[1], gk)
+}
+
+// parseObjFields returns the four parsed object metadata fields:
+// namespace, name, group, kind; or an error if we could not parse
+// the passed string. The name field could possibly contain double
+// underscores, which represent colons. Only the name field can
+// contain the transcoded colon. Because of this, the name field
+// must be parsed last.
+func parseObjFields(s string) ([]string, error) {
```

Comment:
Can this return a struct instead of slice of strings.

Reply:
The struct is created in the function that calls `parseObjFields`.

Reply:
Fixed by combining two functions into one that returns struct. Please have a look.